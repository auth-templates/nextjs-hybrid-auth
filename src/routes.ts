export const PrivateRoutes = {
    dashboard: '/dashboard',
    account: '/account',
    leaves: '/leaves'
};

export const PublicRoutes = {
    login: '/login',
    register: '/register',
    recoverPassword: '/recover-password',
    resendConfirmationEmail: '/resend-confirmation-email',
    resetPassword: '/reset-password/:token',
    verifyAccount: '/verify-account/:token',
    twoStep: '/two-step',
}