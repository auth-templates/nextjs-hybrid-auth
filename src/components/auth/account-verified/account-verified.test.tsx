import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import AccountVerified from './account-verified';
import { PublicRoutes } from '../../routes';

describe("AccountVerified", () => {
    it('should contain all elements', async () => {
        render(
            <MemoryRouter>
                <AccountVerified />
            </MemoryRouter>
        );

        expect(screen.getByText(/Your account has been verified successfully. Please go to/)).toBeInTheDocument(); 
        expect(screen.getByRole('link', { name: 'Sign in' })).toHaveAttribute('href', PublicRoutes.login);
        expect(screen.getByText(/page to sign in./)).toBeInTheDocument();
    });

    it("renders correctly", () => {
        const tree = renderer.create(
            <MemoryRouter>
                <AccountVerified/> 
            </MemoryRouter>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});