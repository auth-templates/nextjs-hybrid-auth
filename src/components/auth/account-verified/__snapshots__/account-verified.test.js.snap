// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountVerified renders correctly 1`] = `
<div
  className="accountVerified"
>
  Your account has been verified successfully. Please go to
  <a
    href="/login"
    onClick={[Function]}
  >
     Sign in 
  </a>
  page to sign in.
</div>
`;
